### ISSUE TITLE
plot(section) should be able to plot all data

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1878]
Users should be able to plot
* In-situ temperature, with `which="temperature"`
* UNESCO-style potential temperature,  with `which="theta"`
* Conservative Temperature, with `which="Conservative Temperature"` or `"CT"`. (Note the upper-case for these GSW/TEOS10 items.)
* Absolute Salinity, with `which="Absolute Salinity"` or `"SA"`. (Note the upper-case for these GSW/TEOS10 items.)
* Anything else contained in the data of the individual stations, or that can be computed from those contents.

**Comment by dankelley** [comment]
#1891 and #1893 having paved the way, I'll likely tackle this within a few days.  It will go into the "develop" branch because, with the foundation I have now, the fix won't be complicated.

**Comment by dankelley** [comment]
This is coming along (not pushed to GH yet -- perhaps later today).  A checklist for my pre-push workplan:

- [ ] fails for unknown (code)
- [ ] fails for unknown (test)
- [ ] map works
- [ ] data works
- [ ] special labelling for CT
- [ ] special labelling for "conservative temperature"
- [ ] special labelling for SA
- [ ] special labelling for Sstar (if any exists)
- [ ] special labelling for "Absolute Salinity"
- [ ] special labelling for "theta"
- [ ] special labelling for "potential temperature"
- [ ] special labelling for sigmaTheta
- [ ] special labelling for N2

**Comment by dankelley** [comment]
This work has me looking at the existing code, which can be made less repetitive with this new approach.

**Comment by dankelley** [comment]
I think I have this working but am still testing and might not be done until next week.  A cute thing, though -- since `pressure` is in each CTD station of `data(section)`, we can plot it.  Below is the result.  It's a nice demonstration, for classes, that pressure and depth are nearly the same ... but not totally the same 😜 

![image](https://user-images.githubusercontent.com/99469/147829704-2b083e7d-aefd-434d-af1b-f537c3c7bf71.png)

**Comment by dankelley** [comment]
This is coming along; "develop" commit  baf04c064d3631fa002ccb6b3dab3acbba31574b produces as below.

``` r
library(oce)
#> Loading required package: gsw
data(section)
section <- subset(handleFlags(section), longitude > (-40))

par(mfrow=c(3,2))
for (which in c("salinity", "temperature", "SA", "CT")) {
    plot(section, which=which)
    mtext(paste0("which=\"",which,"\""), cex=par("cex"))
}
hist(section[["SA"]] - section[["salinity"]],
    main="SA-salinity contrast")
hist(section[["CT"]] - section[["temperature"]],
    main="CT-temperature contrast")
```

![](https://i.imgur.com/eEA1EC9.png)

``` r
available <- section[["?"]]
names <- sort(c(available$data, available$dataDerived))
for (name in names) {
    plot(section, which=name)
    mtext(paste0("which=\"", name, "\""), line=0.5, cex=par("cex"))
}
```

![](https://i.imgur.com/MqB5U1z.png)![](https://i.imgur.com/ujZcCzN.png)![](https://i.imgur.com/Ua6tVU7.png)![](https://i.imgur.com/Vxr7ZSc.png)![](https://i.imgur.com/APpIcRk.png)![](https://i.imgur.com/8WTz2CS.png)

<sup>Created on 2022-01-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by dankelley** [comment]
This is fixed in the bbaadfa9b8a52d1f6435d878eaa6d512f4a14e66  commit of the "develop" branch.
